src/taxonomy_router.rs

```diff
@@ -0,0 +1,246 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+extern crate serde_json;
+
+use controller::Controller;
+use foxbox_adapters::manager::WatchGuard;
+use foxbox_taxonomy::api::{ API, TargetMap };
+use foxbox_taxonomy::values::Value;
+use foxbox_taxonomy::selector::{ GetterSelector, ServiceSelector, SetterSelector };
+use foxbox_taxonomy::services::TagId;
+use foxbox_taxonomy::util::Id;
+use foxbox_users::AuthEndpoint;
+use iron::{ Handler, IronResult, Request, Response };
+use iron::headers::ContentType;
+use iron::method::Method;
+use iron::prelude::Chain;
+use iron::status::Status;
+use serde::Serialize;
+use std::io::Read;
+use std::sync::Mutex;
+
+// This is a specialized Router for the taxonomy API.
+pub struct TaxonomyRouter<A> {
+    api: Mutex<A>
```

Comment:
Can you add a comment here (and possibly open an issue) regarding the performance issue?